website/src/pages/dictionary/index.js

```diff
@@ -72,63 +81,62 @@ function DataDictionary() {
-    ) : (
-      <DnaLoader />
-    );
-
   const context = useDocusaurusContext();
   const {
     siteConfig: {
       customFields: { platformUrl = '' },
     },
   } = context;
 
-  const schemas = get(dictionary, 'schemas', []);
-  const files = schemas.length;
-  const fields = schemas.reduce((acc, schema) => acc + schema.fields.length, 0);
-  const schemaFields = flatten(schemas.map(schema => schema.fields));
-  const { validDataTiers, validDataAttributes } = schemaFields.reduce(
-    (acc, field) => {
-      const meta = get(field, 'meta', {});
-      const { primaryId = false, core = false, dependsOn = false } = meta;
-      const restrictions = get(field, 'restrictions', false);
-      if (primaryId) {
-        acc.validDataTiers.add(TAG_TYPES.id);
-      }
-
-      if (!!restrictions) {
-        acc.validDataAttributes.add(TAG_TYPES.required);
-      }
-
-      if (dependsOn) {
-        acc.validDataAttributes.add(TAG_TYPES.dependency);
-      }
-
-      if (core) {
-        acc.validDataTiers.add(TAG_TYPES.core);
-      }
-
-      if (!core && !primaryId) {
-        acc.validDataTiers.add(TAG_TYPES.extended);
-      }
-      return acc;
-    },
-    { validDataTiers: new Set(), validDataAttributes: new Set() },
-  );
-
   // menu
   const schemaRefs = dictionary.schemas.reduce((acc, schema) => {
     acc[camelCase(schema.name)] = createRef();
     return acc;
   }, {});
-  const menuContents = schemas.map(schema => ({
+  const menuContents = dictionary.schemas.map(schema => ({
     name: startCase(schema.name),
     contentRef: schemaRefs[camelCase(schema.name)],
   }));
 
+  useEffect(() => {
+    const schemas = get(dictionary, 'schemas', []);
+    const files = schemas.length;
+    const fields = schemas.reduce((acc, schema) => acc + schema.fields.length, 0);
+    setMeta({ fileCount: files, fieldCount: fields });
+
+    const schemaFields = flatten(schemas.map(schema => schema.fields));
+    const { validDataTiers, validDataAttributes } = schemaFields.reduce(
+      (acc, field) => {
+        const meta = get(field, 'meta', {});
+        const { primaryId = false, core = false, dependsOn = false } = meta;
+        const restrictions = get(field, 'restrictions', false);
+        if (primaryId) {
+          acc.validDataTiers.add(TAG_TYPES.id);
+        }
+
+        if (!!restrictions) {
+          acc.validDataAttributes.add(TAG_TYPES.required);
+        }
+
+        if (dependsOn) {
+          acc.validDataAttributes.add(TAG_TYPES.dependency);
+        }
+
+        if (core) {
+          acc.validDataTiers.add(TAG_TYPES.core);
+        }
+
+        if (!core && !primaryId) {
+          acc.validDataTiers.add(TAG_TYPES.extended);
+        }
+        return acc;
+      },
+      { validDataTiers: new Set(), validDataAttributes: new Set() },
+    );
+
+    setFilters({ tiers: [...validDataTiers], attributes: [...validDataAttributes] });
+  }, [dictionary]);
```

Comment:
If prevention of unnecessary rerender is the only purpose for this side effects and the filter state, it's better to achieve this with `useMemo` :)